homeassistant/helpers/entity_values.py

```diff
@@ -0,0 +1,42 @@
+"""A class to hold entity values."""
+import fnmatch
+import re
+
+from homeassistant.core import split_entity_id
+
+
+class EntityValues(object):
+    """Class to store entity id based values."""
+
+    def __init__(self, exact=None, domain=None, glob=None):
+        """Initialize an EntityConfigDict."""
+        self._cache = {}
+        self._exact = exact
+        self._domain = domain
+
+        if glob is not None:
+            glob = {re.compile(fnmatch.translate(key)): value
+                    for key, value in glob.items()}
+
+        self._glob = glob
+
+    def get(self, entity_id):
```

Comment:
We will not support deep merging. We will only overwrite keys.

We're not going to support a hack to support another hack. People introduce hack for some specific use case. Then they stop using the hack and guess what? We are left with that hack and have to maintain it forever. Eventually all these hacks will come and kill maintainability, death by a 1000 cuts.

This is going to be the final version of customize. No more changes after this.